modules/transcription-service-microsoft-azure/src/main/java/org/opencastproject/transcription/microsoftazure/MicrosoftAzureTranscriptionService.java

```diff
@@ -520,12 +544,15 @@ void recognizeContinuous(SpeechRecognizer speechRecognizer, String jobId, String
     speechRecognizer.recognized.addEventListener((s, e) -> {
       if (ResultReason.RecognizedSpeech == e.getResult().getReason() && e.getResult().getText().length() > 0) {
         sequenceNumber[0]++;
-        String text = captionFromSpeechRecognitionResult(sequenceNumber[0], e.getResult());
+        List<String> texts = captionFromSpeechRecognitionResult(sequenceNumber[0], e.getResult());
+        sequenceNumber[0] += texts.size() - 1;
 
-        try {
-          writeToTmpFile(text, jobId);
-        } catch (IOException ioException) {
-          logger.error(String.format("Could not write to tmp file: %s", text));
+        for (String text : texts) {
+          try {
+            writeToTmpFile(text, jobId);
+          } catch (IOException ioException) {
+            logger.error(String.format("Could not write to tmp file: %s", text));
```

Comment:
Don't use `format` in a logger statement
```suggestion
            logger.error("Could not write to tmp file: {}", text);
```

Reply:
Thanks!